File: backend/src/main/java/com/erp/timetable/TimetableService.java
package com.erp.timetable;

import com.erp.timetable.dto.TimetableRequest;

import java.util.List;

public interface TimetableService {
    List<Timetable> getTimetableByCourseAndSemester(Long courseId, Integer semester);
    List<Timetable> getTimetableByStudent(Long studentId);
    List<Timetable> getTimetableByStudentAndDate(Long studentId, java.time.LocalDate date);
    Timetable createTimetable(TimetableRequest request);
    Timetable updateTimetable(Long id, TimetableRequest request);
    List<Timetable> getAllTimetables();
    void deleteTimetable(Long id);
}

File: frontend/src/services/timetableService.js
import api from './api';

const timetableService = {
    getTimetableByStudent: async (studentId) => {
        const response = await api.get(`/timetable/student/${studentId}`);
        return response.data;
    },
    getTimetableByStudentAndDate: async (studentId, date) => {
        const response = await api.get(`/timetable/student/${studentId}/today?date=${date}`);
        return response.data;
    }
};

export default timetableService;
